nengo/connection.py

```diff
@@ -119,14 +119,16 @@ class ConnectionFunctionParam(FunctionParam):
 
     def __set__(self, conn, function):
         if function is None:
-            function_info = self.Info(function=None, size=None)
+            function_info = FunctionInfo(function=None, size=None)
```

Comment:
It needs to be pickle-able and for that it needs to be at the top-level of the module.